storm-core/src/jvm/org/apache/storm/metric/internal/CountStatAndMetric.java

```diff
@@ -197,12 +196,6 @@ long readApproximateTime(long value, long timeSpent, long[] bucketTime, long[] b
         return total;
     }
 
-    public void close() {
-        if (_task != null) {
-            _task.cancel();
```

Comment:
If we don't cancel it we leak memory and other things in the tests.  We at least need to close the stat when we are all done, not after calling render-stats.